samples/app-link-unfurling-in-share-to-teams/nodejs/bots/teamsBot.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TeamsActivityHandler, CardFactory } = require("botbuilder");
+
+class TeamsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+    }
+
+    // Invoked when an app based link query activity is received from the connector.
+    async handleTeamsAppBasedLinkQuery(context, query) {
+        var userCard = CardFactory.adaptiveCard(this.getLinkUnfurlingCard());
+        const preview = CardFactory.thumbnailCard(
+            'Adaptive Card',
+            'Please select to get the card'
+            );
+
+        return {
+            composeExtension: {
+                attachmentLayout: 'list',
+                type: 'result',
+                attachments: [{...userCard,preview}]
+            }
+        }
+    }
+
+    // Adaptive card for link unfurling.
+    getLinkUnfurlingCard = () => {
+        var card = {
+          "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+          "type": "AdaptiveCard",
+          "version": "1.4",
+          "body": [
+            {
+              "type": "TextBlock",
+              "size": "Medium",
+              "weight": "Bolder",
+              "text": "Analytics details:"
+            },
+            {
+              "type": "Image",
+              "url": `${process.env.ApplicationBaseUrl}/Images/report.png`
+            }
+          ],
+          "actions":[
+            {
+                "type": "Action.OpenUrl",
+                "title": "Open tab",
+                "url": `https://teams.microsoft.com/l/entity/${process.env.MicrosoftAppId}/tab?webUrl=${process.env.ApplicationBaseUrl}/AuthTab`
+              }
+          ]
```

Comment:
Can we include another action to open this in a stage view? (Sample copied from doc https://docs.microsoft.com/en-us/microsoftteams/platform/tabs/tabs-link-unfurling#:~:text=through%20deep%20link-,Invoke%20Stage%20View%20from%20Adaptive%20Card,the%20stage%20as%20a%20tab.)
`
    type: "Action.Submit",
    name: "View",
    data: {
          msteams: {
            type: "invoke",
            value: {
                type: "tab/tabInfoAction",
                tabInfo: {
                    contentUrl: contentUrl,
                    websiteUrl: websiteUrl,
                    name: "Tasks",
                    entityId: "entityId"
                 }
                }
            }
        }
} `

Reply:
Added button to open tab in stage view